src/components/app/data/hooks/useAlgoliaSearch.ts

```diff
@@ -189,30 +191,35 @@ const useAlgoliaSearch = (indexName: string | null = null): AlgoliaWithCatalogFi
     isIndexSupported,
   } = useSecuredAlgoliaMetadata(indexName);
 
-  const shouldUseSecuredAlgoliaApiKey = (
-    isCatalogQueryFiltersEnabled
-    && isIndexSupported
-    && !!securedAlgoliaMetadata.securedAlgoliaApiKey
-  );
-
-  // Based on the waffle flag and supported indexes, we will use the secured algolia
-  // key or default back to the legacy initialization of the search client and indexes
-  // The fallback once the secured algolia api key has been rolled out to all users via a waffle flag
-  // (catalogQuerySearchFiltersEnabled) is considered an errored state where downstream components would
-  // display the <SearchUnavailableAlert /> if no search client is returned or the upstream secured
-  // algolia api call fails from the BFF.
-  const algoliaSearchApiKey: string = shouldUseSecuredAlgoliaApiKey
-    ? securedAlgoliaMetadata.securedAlgoliaApiKey!
-    : config.ALGOLIA_SEARCH_API_KEY;
-
   // Update instantiate search client with or without a secured
   // algolia api key and retrieve the initialized algolia index
   return useMemo(() => {
+    const shouldUseSecuredAlgoliaApiKey = (
+      isCatalogQueryFiltersEnabled
+      && isIndexSupported
+      && !!securedAlgoliaMetadata.securedAlgoliaApiKey
+    );
+
+    const hasCatalogUuidToCatalogQueryUuidMapping = !isObjEmpty(
+      securedAlgoliaMetadata.catalogUuidsToCatalogQueryUuids,
+    );
+
+    // Based on the waffle flag and supported indexes, we will use the secured algolia
+    // key or default back to the legacy initialization of the search client and indexes
+    // The fallback once the secured algolia api key has been rolled out to all users via a waffle flag
+    // (catalogQuerySearchFiltersEnabled) is considered an errored state where downstream components would
+    // display the <SearchUnavailableAlert /> if no search client is returned or the upstream secured
+    // algolia api call fails from the BFF.
+    const algoliaSearchApiKey: string = shouldUseSecuredAlgoliaApiKey && hasCatalogUuidToCatalogQueryUuidMapping
+      ? securedAlgoliaMetadata.securedAlgoliaApiKey!
+      : config.ALGOLIA_SEARCH_API_KEY;
```

Comment:
Note: there's a related comment suggesting to rely on TypeScript inference here of the return values vs. explicitly defining the `: string` type here. I believe it's partially obscuring the fact that `algoliaSearchApiKey` may not actually exist, with `config.ALGOLIA_SEARCH_API_KEY=null`.